typescript/src/schema/proto.ts

```diff
@@ -703,19 +701,20 @@ export function getSingleRoutingHeaderParam(
     dynamicRoutingRule = {
       fieldRetrieve: convertFieldToCamelCase(rule.field),
       fieldSend: rule.field,
-      messageRegex: '[^/]+',
-      namedSegment: '[^/]+',
+      messageRegex: `(?<${rule.field}>.*)`,
     };
   }
   // If the annotation is malformed, then return empty interface
-  else if (getNamedSegment(rule.pathTemplate).length < 1) {
-    return dynamicRoutingRule;
-  } else {
+  else {
```

Comment:
I assume that if the annotation is malformed, those values don't get appended to the header. What happens to the request in that case?

Reply:
At this moment, only two APIs (BigTable and CloudRun) use this functionality, and I checked that everything works :)

Reply:
Most likely it will fail on the backend. The client library is not supposed to validate the request, it's OK to let it through and let the backend respond.